pytest_selenium/drivers/saucelabs.py

```diff
@@ -71,27 +67,29 @@ def driver_kwargs(request, test, capabilities, **kwargs):
     tags = capabilities.get('tags', []) + markers
     if tags:
         capabilities['tags'] = tags
-    executor = EXECUTOR_URL.format(
-        username=_username(request.config),
-        key=_api_key(request.config))
+    executor = EXECUTOR_URL.format(**credentials())
     kwargs = {
         'command_executor': executor,
         'desired_capabilities': capabilities}
     return kwargs
 
 
-def _api_key(config):
-    api_key = config.getini('sauce_labs_api_key')
-    if not api_key:
-        raise pytest.UsageError('Sauce Labs API key must be set')
-    return api_key
-
-
-def _username(config):
-    username = config.getini('sauce_labs_username')
+def credentials():
```

Comment:
Perhaps it would be worth refactoring `credentials` so it can be reused by the various drivers? The logic seems to be identical across them: read a config file, try to get the credentials from it, fallback to an environment variable, etc.